dagmc_stats.py

```diff
@@ -55,44 +61,147 @@ def get_native_ranges(my_core, meshset, entity_types):
+    return t_p_v_data
+  
+  
+def get_triangles_per_surface(my_core, entity_ranges):
+    """
+    This function will return data about the number of triangles on each
+    vertex in a file
+    
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    entity_ranges : a dictionary containing ranges for each type in the file 
+                    (VOLUME, SURFACE, CURVE, VERTEX, TRIANGLE, ENTITYSET)
+    
+    outputs
+    -------
+    t_p_s : a dictionary containing the entityhandle of the surface,
+            and the number of triangles each surface contains.
+            i.e {surface entityhandle : triangles it contains}
+    """
+
+    t_p_s = {}
+    for surface in entity_ranges['Surfaces']:
+        t_p_s[surface] = my_core.get_entities_by_type(
+                                 surface, types.MBTRI).size()
+    return t_p_s
+
+  
 def get_surfaces_per_volume(my_core, entityset_ranges):
     """
     Get the number of surfaces that each volume in a given file contains
-
+    
     inputs
     ------
     my_core : a MOAB core instance
     entity_ranges : a dictionary of the entityset ranges of each tag in a file
-
+    
     outputs
     -------
-    freqs : a list of the number of surfaces each volume in the file contains
-
+    s_p_v : a dictionary containing the volume entityhandle
+            and the number of surfaces each volume in the file contains
+            i.e. {volume entityhandle:surfaces it contains}
     """
 
-    s_p_v = np.array([])
+    s_p_v = {}
     for volumeset in entityset_ranges['Volumes']:
-        s_p_v = np.append(s_p_v, my_core.get_child_meshsets(volumeset).size())
+        s_p_v[volumeset] = my_core.get_child_meshsets(volumeset).size()
     return s_p_v
+
+
+def get_triangle_aspect_ratio(my_core, meshset):
+    """
+    Gets the triangle aspect ratio (according to the equation: (abc)/(8(s-a)(s-b)(s-c)), where s = .5(a+b+c).)
+    
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    entityset_ranges : a dictionary with one entry for each entityset type, and the value is the range of entities that corrospond to 
+                        each type
+    meshset : a meshset containing a certain part of the mesh
+    
+    outputs
+    -------
+    t_a_r : (list) the triangle aspect ratios for all triangles in the meshset
+    """
+    
+    t_a_r = []
+    indexes = [1,2,0]
+    
+    tris = my_core.get_entities_by_type(meshset, types.MBTRI)
+    
+    for triangle in tris:
+        side_lengths = []
+        s = 0
+        coord_list = []
+        
+        verts = list(my_core.get_adjacencies(triangle, 0))
+        
+        for side in range(3):
+            coords = list(my_core.get_coords(verts[side]))
+            coord_list.append(coords)
+            
+        coord_array = np.array(coord_list)
+        
+        for length in range(3):    
+            sum_squares = 0
+            
+            for vert in range(3):
```

Comment:
With the right data structure, you should be able to do this as a numpy vector operation.

See this, for example: https://stackoverflow.com/questions/1401712/how-can-the-euclidean-distance-be-calculated-with-numpy

Reply:
This works very nicely with a list of arrays